spmd/tensor/ops/matrix_ops.py

```diff
@@ -35,31 +36,37 @@ def dist_addmm(
     )
     device_mesh = mat1.device_mesh
 
-    assert input_placement.is_replicate(), "only support replication now"
-
-    # only implemented combo with no comm for now
-    # TODO: implement all combinations
-    if mat1_placement.is_shard(dim=0) and mat2_placement.is_replicate():
-        local_res = local_input.addmm(
-            local_mat1, local_mat2, beta=beta, alpha=alpha
+    if (
+        (
+            input_placement.is_replicate()
+            and mat1_placement.is_replicate()
+            and mat2_placement.is_replicate()
         )
-        return Tensor.from_local(local_res, device_mesh, mat1.placements)
-    elif mat1_placement.is_replicate() and mat2_placement.is_shard(dim=1):
-        local_res = local_input.addmm(
-            local_mat1, local_mat2, beta=beta, alpha=alpha
+        or (
+            input_placement.is_shard(dim=0)
+            and mat1_placement.is_shard(dim=0)
+            and mat2_placement.is_replicate()
         )
-        return Tensor.from_local(local_res, device_mesh, mat2.placements)
-    elif mat1_placement.is_replicate() and mat2_placement.is_replicate():
-        local_res = local_input.addmm(
-            local_mat1, local_mat2, beta=beta, alpha=alpha
+        or (
+            input_placement.is_shard(dim=1)
+            and mat1_placement.is_replicate()
+            and mat2_placement.is_shard(dim=1)
         )
-        return Tensor.from_local(
-            local_res, device_mesh, mat1.placements, run_check=False
+        or (
+            input_placement.is_partial()
+            and mat1_placement.is_shard(dim=1)
+            and mat2_placement.is_shard(dim=0)
         )
-    else:
-        raise RuntimeError(
-            f"addmm operator supported for inputs: {mat1}, {mat2}"
+    ):
+        local_res = local_input.addmm(
+            local_mat1, local_mat2, beta=beta, alpha=alpha
         )
+        return Tensor.from_local(local_res, device_mesh, input.placements)
```

Comment:
Took a closer look, there're some issues here, we cannot simply assume the output placement is the same in every case, in fact, in multiple cases, they are different, we need to specify the output placement case by case.

Reply:
as a reference, please see P519128154 for a correct output placement specification